File: hpgl2eps.js
const fs = require('fs');

function convertHPGLToEPS(inputPath, outputPath) {
    const inputData = fs.readFileSync(inputPath, 'utf8');
    const lines = inputData.split('\n');
    let outputData = "%!PS-Adobe-3.0 EPSF-3.0\n";
    outputData += "%%BoundingBox: 0 0 100 100\n";
    outputData += "0.2662068576 0.2662068576 scale\n";
    outputData += "newpath\n";

    lines.forEach(line => {
        const commands = line.split(';');
        commands.forEach(command => {
            const postScriptCommand = convertHPGLCommandToPostScript(command);
            if (postScriptCommand) {
                outputData += postScriptCommand + "\n";
            }
        });
    });

    outputData += "stroke\n";
    outputData += "showpage\n";

    fs.writeFileSync(outputPath, outputData, 'utf8');
}

function convertHPGLCommandToPostScript(hpglCommand) {
    if (hpglCommand.startsWith("PU")) {
        return handleCoordinates(hpglCommand.substring(2), "moveto");
    } else if (hpglCommand.startsWith("PD")) {
        return handleCoordinates(hpglCommand.substring(2), "lineto");
    } else if (hpglCommand.startsWith("PA")) {
        return handleCoordinates(hpglCommand.substring(2), "moveto");
    }
    return null;
}

function handleCoordinates(coordinateString, command) {
    const coordinates = coordinateString.split(',');
    if (coordinates.length === 2) {
        const scaleFactor = 0.2662068576;
        const x = parseFloat(coordinates[0]) * scaleFactor;
        const y = parseFloat(coordinates[1]) * scaleFactor;
        return `${x} ${y} ${command}`;
    }
    return null;
}

const inputPath = 'input.hpgl';
const outputPath = 'output.eps';
convertHPGLToEPS(inputPath, outputPath);

File: hpgl2eps.cs
using System;
using System.IO;
using System.Globalization;

class HPGLToEPSConverter
{
    static void Main()
    {
        string inputPath = "input.hpgl";
        string outputPath = "output.eps";

        ConvertHPGLToEPS(inputPath, outputPath);
    }

    static void ConvertHPGLToEPS(string inputPath, string outputPath)
    {
        using (StreamReader reader = new StreamReader(inputPath))
        using (StreamWriter writer = new StreamWriter(outputPath))
        {
            
            writer.WriteLine("%!PS-Adobe-3.0 EPSF-3.0");
            writer.WriteLine("%%BoundingBox: 0 0 100 100");
            writer.WriteLine("0.2662068576 0.2662068576 scale");

            writer.WriteLine("newpath");

            string line;
            while ((line = reader.ReadLine()) != null)
            {
                string[] commands = line.Split(';');
                foreach (var command in commands)
                {
                    string postScriptCommand = ConvertHPGLCommandToPostScript(command);
                    if (!string.IsNullOrEmpty(postScriptCommand))
                    {
                        writer.WriteLine(postScriptCommand);
                    }
                }
            }

            writer.WriteLine("stroke");
            writer.WriteLine("showpage");
        }
    }

    static string ConvertHPGLCommandToPostScript(string hpglCommand)
    {
        if (hpglCommand.StartsWith("PU"))
        {
            return HandleCoordinates(hpglCommand.Substring(2), "moveto");
        }
        else if (hpglCommand.StartsWith("PD"))
        {
            return HandleCoordinates(hpglCommand.Substring(2), "lineto");
        }
        else if (hpglCommand.StartsWith("PA"))
        {
            return HandleCoordinates(hpglCommand.Substring(2), "moveto");
        }
        return null;
    }

    static string HandleCoordinates(string coordinateString, string command)
    {
        var coordinates = coordinateString.Split(',');
        if (coordinates.Length == 2)
        {
            double scaleFactor = 0.2662068576;
            double x = double.Parse(coordinates[0], CultureInfo.InvariantCulture);
            double y = double.Parse(coordinates[1], CultureInfo.InvariantCulture);
            return $"{x} {y} {command}";
        }
        return null;
    }
}
